api/main_endpoints/models/Course.js

```diff
@@ -0,0 +1,27 @@
+const mongoose = require("mongoose");
+const Schema = mongoose.Schema;
+
+const CourseSchema = new Schema(
+  {
+    title: {
+      type: String,
+      required: true
+    },
+    author: {
+      type: String
+    },
+    description: {
+      type: String
+    },
+    lessons: {
+      type: Array
```

Comment:
There is a better way to have a embedded document field. 

Link: https://stackoverflow.com/questions/43429500/how-do-i-properly-embed-documents-in-mongoose

This is better because it will store an array of Lesson ObjectId's. You will also need to change your api-tests.